compiler/typecheck/TcSplice.hs

```diff
@@ -1382,7 +1382,7 @@ reifyTyCon tc
                    Just name ->
                      let thName   = reifyName name
                          injAnnot = tyConInjectivityInfo tc
-                         sig = TH.TyVarSig (TH.KindedTV thName kind')
+                         sig = TH.TyVarSig (TH.KindedTV thName kind' TH.Specified) -- TODO: I *think* Specified is right here...
```

Comment:
Using `HsTyVarBndr` in result sigs is a lie. I would favor `Required`, though.

Reply:
What is a result sig? Do you think the change to get rid of the lie would be very far-reaching?

Reply:
It wouldn't be far-reaching, but it would be annoying. A result sig is part of the injective type families implementation. (Check the paper or the manual for more info.) I favor retaining the lie.